### PR TITLE
Fix tmp directories

### LINKED ISSUE #636: [Bug]: Change permissions and ownership for the temp directories to be the same as the parent directory
### What happened?

When checking disk usage, errors like `du: cannot read directory 'ac.benedict/E3SMv3_dev/20231012.v3alpha04_bigrid_L80_QBO1.F20TR.chrysalis/post/scripts/tmp.407421.rCki': Permission denied`. This is because of a permissions issues with the temp directories `zppy` creates.

### What machine were you running on?

Chrysalis

### Environment

N/A

### What command did you run?

```bash
du
```

### Copy your cfg file

```bash
N/A
```

### What jobs are failing?

_No response_

### What stack trace are you encountering?

_No response_

### PR DESCRIPTION
**Pull request by forsyth2**
## Issue resolution
- Closes #636

Select one: This pull request is...
- [x] a bug fix: increment the patch version
- [ ] a small improvement: increment the minor version
- [ ] an incompatible (non-backwards compatible) API change: increment the major version

## 1. Does this do what we want it to do?

Objectives:
- Fix temp directory permissions
- Give temp directories more descriptive names

Required:
- [ ] Product Management: I have confirmed with the stakeholders that the objectives above are correct and complete.
- [ ] Testing: I have added or modified at least one "min-case" configuration file to test this change. Every objective above is represented in at least one cfg.
- [ ] Testing: I have considered likely and/or severe edge cases and have included them in testing.

If applicable:
- [ ] Testing: this pull request introduces an important feature or bug fix that we _must_ test often. I have updated the weekly-test configuration files, not just a "min-case" one.
- [ ] Testing: this pull request adds at least one new possible parameter to the cfg. I have tested using this parameter with and without any other parameter that may interact with it.

## 2. Are the implementation details accurate & efficient?

Required:
- [ ] Logic: I have visually inspected the entire pull request myself.
- [ ] Logic: I have left GitHub comments highlighting important pieces of code logic. I have had these code blocks reviewed by at least one other team member.

If applicable:
- [ ] Dependencies: This pull request introduces a new dependency. I have discussed this requirement with at least one other team member. The dependency is noted in `zppy/conda`, not just an `import` statement.

## 3. Is this well documented?

Required:
- [ ] Documentation: by looking at the docs, a new user could easily understand the functionality introduced by this pull request.

## 4. Is this code clean?

Required:
- [ ] Readability: The code is as simple as possible and well-commented, such that a new team member could understand what's happening.
- [ ] Pre-commit checks: All the pre-commits checks have passed.

If applicable:
- [ ] Software architecture: I have discussed relevant trade-offs in design decisions with at least one other team member. It is unlikely that this pull request will increase tech debt.

### CODE REVIEW

File: zppy/templates/e3sm_diags.bash [hunk 1]
```diff
@@ -40,7 +40,8 @@ tag="{{ tag }}"
 results_dir=${tag}_${Y1}-${Y2}
 
 # Create temporary workdir
-workdir=`mktemp -d tmp.${id}.XXXX`
+hash=`mktemp --dry-run -d XXXX`
+workdir=`mkdir tmp.{{ prefix }}.${id}.${hash}`
```
Comment (forsyth2) [inline zppy/templates/e3sm_diags.bash:44]: I'm getting errors in `/lcrc/group/e3sm/ac.forsyth2/zppy_min_case_nco_output/test-636-20241209/v3.LR.historical_0051/post/scripts`, so this can't be sufficient. Will need to debug more what's going on.

**Comment by forsyth2** [comment]
For reference, as of this PR:
- bash files have permissions:           `-rwxrw----`
- Other files have permissions:          `-rw-rw----`
- tmp directories have permissions: `drwxrws---`
